test/mocha/safari9.js

```diff
@@ -0,0 +1,26 @@
+var assert = require('assert');
+var UglifyJS = require('../..');
+
+describe('legacy safari', function() {
+    it('Should make sure mangled function name is different from its argument names', function () {
+        assert.strictEqual(
+            UglifyJS.minify([
+                'function main() {',
+                '    "use strict";',
+                '    ',
+                '    function n(a,b,c) {',
+                '        console.log(c)',
+                '    }',
+                '    E.on = function f2() {',
+                '        E.on(function(e) {',
+                '            return n(this,e)',
+                '        })',
+                '    }',
+                '}'
+            ].join('\n'), {
+                ie8: true
+            }).code,
+            'function main(){"use strict";E.on=function(){E.on(function(n){return function c(n,o,t){console.log(t)}()})}}'
+        )
+    })
+})
```

Comment:
Please migrate this test over here, right underneath this one (which also serves as an example for the format)
https://github.com/mishoo/UglifyJS2/blob/fc78423f1d9bfa10774a04563e37e130523bf46d/test/compress/webkit.js#L53-L60